app/javascript/src/components/Dashboard/Contacts/Table.jsx

```diff
@@ -1,53 +1,48 @@
-import React, { useState } from "react";
+import React, { useEffect, useState, useRef } from "react";
 
+import { MenuHorizontal } from "neetoicons";
 import { Table as NeetoUITable } from "neetoui";
 
-import { renderContactDetails, renderText, renderDropdown } from "./utils";
+import { renderDropdown } from "components/Dashboard/utils";
+
+import { GET_COLUMN_DATA } from "./constants";
 
 const Table = ({ contacts = [], onDeleteContact }) => {
   const [pageNumber, setPageNumber] = useState(1);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const timeoutRef = useRef(null);
+
+  const renderIcon = (_, { id }) =>
+    renderDropdown({
+      onDelete: onDeleteContact,
+      onEdit: () => {},
+      id,
+      icon: MenuHorizontal,
+    });
+
+  const columnData = GET_COLUMN_DATA(renderIcon);
+
+  useEffect(() => {
+    timeoutRef.current = setTimeout(() => {
+      setIsLoading(false);
+    }, 2000);
```

Comment:
Try to keep the hooks at the top and rest of the things below it unless the hook depends on the value.